### ISSUE TITLE
3 llvm-test-suite tests failed on GPU with PI CUDA error after pulldown from LLORG

### ISSUE THREAD

**Issue by vmaksimo** [intel/llvm#4079]
3 LLVM test suite tests started to fail on CUDA GPU after LLORG pulldown (https://github.com/intel/llvm/pull/4054). They are temporarily disabled by https://github.com/intel/llvm-test-suite/pull/349 to unlock the pulldown process.

List of failed tests:
``` 
SYCL :: Basic/multisource.cpp, 
SYCL :: SeparateCompile/same-kernel.cpp,
SYCL :: SeparateCompile/sycl-external.cpp
```
Failure log:
```
PI CUDA ERROR:
	Value:           500
	Name:            CUDA_ERROR_NOT_FOUND
	Description:     named symbol not found
	Function:        cuda_piKernelCreate
	Source Location: sycl/plugins/cuda/pi_cuda.cpp:2299

terminate called after throwing an instance of 'cl::sycl::runtime_error'
  what():  Native API failed. Native API returns: -999 (Unknown OpenCL error code) -999 (Unknown OpenCL error code)

error: command failed with exit status: -6
```

**To Reproduce**
``` 
# If the https://github.com/intel/llvm/pull/4054 is not merged yet, fetch the PR:
git fetch origin pull/4054 /head:pr_4054
git checkout pr_4054 
# Otherwise, please checkout the latest sycl branch

# Then build with the normal procedure https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain
python $DPCPP_HOME/llvm/buildbot/configure.py
python $DPCPP_HOME/llvm/buildbot/compile.py

# Get sources of intel/llvm-test-suite
git clone https://github.com/intel/llvm-test-suite

# Then revert the commit that disables the test
# See the hash in PR: https://github.com/intel/llvm-test-suite/pull/349
git revert <hash_placeholder>

# Then build intel/llvm-test-suite per normal procedure
# https://github.com/intel/llvm-test-suite/tree/intel/SYCL#build-and-run-tests

```
**Environment (please complete the following information):**
- OS: Linux
- Target device and vendor: CUDA

**Comment by steffenlarsen** [comment]
Thanks for the detailed report! This problem with https://github.com/intel/llvm/pull/4112 and the corresponding tests are re-enabled for CUDA with https://github.com/intel/llvm-test-suite/pull/365.

**Comment by bader** [comment]
@steffenlarsen, does #4112 resolves https://github.com/intel/llvm/issues/4088 as well?

**Comment by steffenlarsen** [comment]
> 
> 
> @steffenlarsen, does #4112 resolves #4088 as well?

I believe so, but I will have to test it.

**Comment by steffenlarsen** [comment]
@bader I can confirm that #4112 seems to resolve #4088 .
